sql/log.cc

```diff
@@ -7979,14 +8118,12 @@ int Event_log::write_cache(THD *thd, binlog_cache_data *cache_data)
       likely(!crypto.scheme) &&
       likely(!opt_binlog_legacy_event_pos))
   {
-    int res= my_b_copy_all_to_cache(cache, &log_file);
+    int res=
+        my_b_copy_to_cache(cache, &log_file, cache_data->length_for_read());
     status_var_add(thd->status_var.binlog_bytes_written, my_b_tell(cache));
```

Comment:
Apparently `my_b_tell(cache)` the increment to `binlog_bytes_written`  needs adjustment now to subtract `cache_data->length_for_read()` that is not copied to the binlog file.

Reply:
fixed